lib/puppet/file_system/windows.rb

```diff
@@ -177,14 +200,17 @@ def set_dacl(path, dacl)
     Puppet::Util::Windows::Security.set_security_descriptor(path, new_sd)
   end
 
-  def secure_dacl(current_sid)
+  def secure_dacl(current_sid, owner_permission = FULL_CONTROL, group_permission = FULL_CONTROL)
     dacl = Puppet::Util::Windows::AccessControlList.new
     [
      Puppet::Util::Windows::SID::LocalSystem,
      Puppet::Util::Windows::SID::BuiltinAdministrators,
      current_sid
     ].uniq.map do |sid|
       dacl.allow(sid, FULL_CONTROL)
+      permission = owner_permission
+      permission = group_permission if Puppet::Util::Windows::SID::BuiltinAdministrators
+      dacl.allow(sid, permission)
```

Comment:
I'm not sure I follow this. If `sid` is granted `FULL_CONTROL` above on line 210, then this line doesn't grant any additional permissions.

Reply:
Removed `dacl.allow(sid, FULL_CONTROL)`

Reply:
`LocalSystem` and `BuiltinAdministrators` should always have `FULL_CONTROL`, so this method shouldn't change.

Reply:
This means that `current_sid` will always have full control. Looking at `0440` is it worth changing `current_sid` permissions?